Tweaks for prerelease

vaidehijoshi opened the pull request:
- [x] Tweak API for `fadein()` and `fadeout()` operations
- [x] Improve `README`
- [x] Update `Cargo.toml` for `1.0` release

vaidehijoshi committed:
Tweak API for fadein() and fadeout() operations.

Previously, we were returning `self` for these operations when impled on
fully-opaque colors. This changes the API to diverge from Less'
implementation such that, for opaque colors, we convert into the alpha
channel equivalent (rgba or hsla) and then perform the color operation
as expected.

vaidehijoshi committed:
Add in-depth README with examples, links to docs.

vaidehijoshi committed:
Update Cargo.toml for 1.0 release

chancancode commented on README.md:
```diff
@@ -2,6 +2,127 @@
 [![Build Status](https://travis-ci.com/vaidehijoshi/css-colors.svg?branch=master)](https://travis-ci.com/vaidehijoshi/css-colors) [![css-colors](https://docs.rs/css-colors/badge.svg)](https://docs.rs/css-colors)
 
-A Rust parser and converter for CSS colors.
+A Rust parser and converter to transform CSS colors. 🎨
```
:scissors: parser since we didn’t get to that part yet

chancancode commented on README.md:
```diff
@@ -2,6 +2,127 @@
+
+assert_eq!(chartreuse.darken(percent(20), hsl(9, 100, 30));
+assert_eq!(chartreuse.desaturate(percent(20), hsl(9, 80, 50));
+assert_eq!(chartreuse.greyscale(), hsl(90, 0, 50));
```
Missing some parens here

chancancode commented on README.md:
````diff
@@ -2,6 +2,127 @@
+Manipulate single colors to create new color model representations:
+```rust
+use css_colors::{Color, hsl};
+
````
Missing percent import

chancancode commented on README.md:
```diff
@@ -2,6 +2,127 @@
+### Linting + plugins
+
+Please use the below plugins to ensure code consistency when contributing to this crate.
+* [RLS-based linting for VSCode](https://github.com/rust-lang-nursery/rls-vscode)
```
rustfmt

chancancode reviewed (approved):
:tada:

vaidehijoshi committed:
Clean up README & Cargo.toml

Remove references to "parser", format imports in examples correctly,
update required linters/plugins in Contributing section.
